react/Payment.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useRef, useEffect } from 'react'
+
+if (window && window.document) {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  var postRobot = require('post-robot')
+}
+
+interface Card {
+  cardNumber: string
+  cardName: string
+  cardDate: string
+  cardCvv: string
+}
+
+const IFRAME_APP_VERSION = '0.1.1'
+const LOCAL_IFRAME_DEVELOPMENT = false
+
+let iframeURL = `https://io.vtexpayments.com.br/card-form-ui/${IFRAME_APP_VERSION}/index.html`
+
+if (LOCAL_IFRAME_DEVELOPMENT) {
+  const PORT = 3000
+  iframeURL = `https://checkoutio.vtexlocal.com.br:${PORT}/`
+}
+
+const Payment: React.FC = () => {
+  const [iframeData, setIframeData] = useState<Card | null>(null)
+  const iframeRef = useRef<HTMLIFrameElement>(null)
+
+  const setupIframe = () => {
+    const stylesheetsUrls = Array.from(
+      document.head.querySelectorAll<HTMLLinkElement>('link[rel=stylesheet]')
+    ).map(link => link.href)
+
+    postRobot.send(iframeRef.current!.contentWindow, 'setup', {
```

Comment:
Will this work at SSR? `postRobot` is only defined when `window && window.document`, as stated in line 3

Reply:
The method is called when iframe loaded. Look: `onLoad={() => setupIframe()}`